tests/model_registry/conftest.py

```diff
@@ -467,3 +517,481 @@ def is_model_registry_oauth(request: FixtureRequest) -> bool:
 def api_server_url(admin_client: DynamicClient) -> str:
     infrastructure = Infrastructure(client=admin_client, name="cluster", ensure_exists=True)
     return infrastructure.instance.status.apiServerURL
+
+
+# =============================================================================
+# FACTORY FIXTURES
+# =============================================================================
+
+
+@pytest.fixture(scope="class")
+def model_registry_db_factory(
+    admin_client: DynamicClient,
+    pytestconfig: Config,
+    model_registry_cleanup_registry: ModelRegistryCleanupRegistry,
+) -> Generator[Callable[[ModelRegistryDBConfig], ModelRegistryDBBundle], Any, Any]:
+    """Factory fixture for creating Model Registry DB bundles."""
+    created_bundles: List[ModelRegistryDBBundle] = []
+
+    def create_db_bundle(config: ModelRegistryDBConfig) -> ModelRegistryDBBundle:
+        """Create a complete Model Registry DB bundle."""
+        # Generate unique names to avoid conflicts
+        unique_suffix = str(uuid.uuid4())[:8]
+        db_name = f"{config.name_prefix}-{unique_suffix}"
+
+        # Use existing labels helper or create custom ones
+        labels = config.labels or get_model_registry_db_label_dict(db_resource_name=db_name)
+        annotations = config.annotations or MODEL_REGISTRY_DB_SECRET_ANNOTATIONS
+
+        # Handle post-upgrade scenario
+        if pytestconfig.option.post_upgrade:
```

Comment:
I don't like the idea of combining all the resource creation together. This would make triaging difficult.